app/views/directives/create-secret.html

```diff
@@ -47,268 +47,296 @@
+            Optional username for Git authentication.
+          </div>
+        </div>
 
-    <div ng-if="newSecret.authType === 'kubernetes.io/dockercfg'">
-      <div class="form-group" ng-class="{ 'has-error' : secretForm.dockerServer.$invalid && secretForm.dockerServer.$touched }">
-        <label for="dockerServer" class="required">Image Registry Server Address</label>
-        <div>
+        <div class="form-group" ng-class="{ 'has-error' : secretForm.passwordToken.$invalid && secretForm.passwordToken.$touched }">
+          <label ng-class="{ required: !add.cacert && !add.gitconfig }" for="passwordToken">Password or Token</label>
           <input class="form-control"
-            id="dockerServer"
-            name="dockerServer"
-            ng-model="newSecret.data.dockerServer"
-            type="text"
+            id="passwordToken"
+            name="passwordToken"
+            ng-model="newSecret.data.passwordToken"
             autocorrect="off"
             autocapitalize="none"
             spellcheck="false"
-            required>
+            aria-describedby="password-token-help"
+            type="password"
+            ng-required="!add.cacert && !add.gitconfig">
         </div>
-      </div>
-      <div ng-show="secretForm.dockerServer.$error.required && secretForm.dockerServer.$touched" class="has-error">
-        <div class="help-block">
-          Image registry server address is required.
+        <div class="has-error" ng-show="secretForm.passwordToken.$error.required && secretForm.passwordToken.$touched">
+          <div class="help-block">
+            Password or token is required.
+          </div>
+        </div>
+        <div class="help-block" id="password-token-help">
+          Password or token for Git authentication. Required if a ca.crt or .gitconfig file is not specified.
         </div>
-      </div>
 
-      <div class="form-group" ng-class="{ 'has-error' : secretForm.dockerUsername.$invalid && secretForm.dockerUsername.$touched }">
-        <label for="dockerUsername" class="required">Username</label>
-        <div>
-          <input class="form-control"
-            id="dockerUsername"
-            name="dockerUsername"
-            ng-model="newSecret.data.dockerUsername"
-            type="text"
-            autocorrect="off"
-            autocapitalize="none"
-            spellcheck="false"
-            required>
+        <div class="form-group">
+          <div class="checkbox">
+            <label>
+              <input type="checkbox" ng-model="add.cacert">
+              Use a custom ca.crt file
+            </label>
+          </div>
         </div>
-      </div>
-      <div ng-show="secretForm.dockerUsername.$error.required && secretForm.dockerUsername.$touched" class="has-error">
-        <div class="help-block">
-          Username is required.
+        <div class="form-group" ng-if="add.cacert" id="cacert">
+          <label class="required" for="cacert">CA Certificate File</label>
```

Comment:
for attribute value does not match id value of input

Reply:
It is... under the `label` tag there is a `osc-file-input` directive to which we pass appropriate id in `drop-zone-id`